### PR TITLE
cp-170: Add Meditation Player Layout

### PR DESCRIPTION
**Pull request by danilittaa**
- added '@react-native-community/slider' and 'react-native-track-player' libraries;
- created useTrackPlayerProgress and useTrackPlayerControls hooks;
- all functionality is working
- music continues to play even when you exit the application or turn off the phone

image

https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/108797614/41d41bcc-94c8-4a10-8fd3-6df7f3e78d5c

P.S. It would be better if you first merged my other task, 'Meditations List Layout #101', so that I can add navigation to this page

### CODE REVIEW

**Review by iamAlinaaa — DISMISSED** [review]
All looks good ) as i understood you have special bottom tab for this player (meditations) and after previous task will be merged , you add it there ?)

**Comment by OlenaYatseniuk** [comment]
In issue #78, I changed getFormattedTime helper and moved it to shared folder, so you can use it when it is merged.

**Review by OlenaYatseniuk — DISMISSED** [review]
Great job!

File: mobile/src/libs/helpers/get-formatted-time.helper.ts
Comment (v0ldemar01) [inline mobile/src/libs/helpers/get-formatted-time.helper.ts]: Please check implementation on frontend side, your handlers are good but we will have one style for best reusing code)

**Review by v0ldemar01 — CHANGES_REQUESTED** [review]

File: mobile/src/libs/helpers/get-formatted-time.helper.ts
Comment (v0ldemar01) [reply mobile/src/libs/helpers/get-formatted-time.helper.ts]: As @OlenaYatseniuk already mentined)

Comment (danilittaa) [reply mobile/src/libs/helpers/get-formatted-time.helper.ts]: Then I guess I'll wait until this task on the frontend will be merged

Comment (liza-veis) [reply mobile/src/libs/helpers/get-formatted-time.helper.ts]: Merged

File: mobile/index.js
Comment (dmytro-kucherenko) [inline mobile/index.js]: Are imports have right order? I thought `react-native-track-player` should be placed near `react-native`

File: mobile/src/screens/meditation/components/controls/components/button/button/button.tsx
Comment (dmytro-kucherenko) [inline mobile/src/screens/meditation/components/controls/components/button/button/button.tsx]: It can be misunderstanding that `isRounded` flag change the color of the icon. Also we already have common component with `Button` name. If it's really necessary to make separate component, maybe it's better to name it like `MeditationControlButton` or smth like that?

Comment (danilittaa) [reply mobile/src/screens/meditation/components/controls/components/button/button/button.tsx]: isRounded changes the color because this way I indicate that this button is central (in the design, it's blue).
I don't see the point in complicating component names

**Review by BrianSammit — DISMISSED** [review]

File: mobile/src/screens/meditation-list/components/meditation-item/meditation-item.tsx
Comment (v0ldemar01) [inline mobile/src/screens/meditation-list/components/meditation-item/meditation-item.tsx]: Do we already have an enum for it?

**Review by OlesiaTr — DISMISSED** [review]

**Review by iamAlinaaa — DISMISSED** [review]

File: mobile/src/libs/hooks/use-track-player-progress/use-track-player-progress.hook.ts
Comment (dmytro-kucherenko) [inline mobile/src/libs/hooks/use-track-player-progress/use-track-player-progress.hook.ts]: According to the rule B16 we shouldn't use abbreviations and similar word forms

File: mobile/src/libs/hooks/use-track-player-controls/use-track-player-controls.hook.ts
Comment (v0ldemar01) [inline mobile/src/libs/hooks/use-track-player-controls/use-track-player-controls.hook.ts]: Please rename due to quality criteria rule

File: mobile/src/screens/meditation/components/controls/components/button/button/button.tsx
Comment (v0ldemar01) [inline mobile/src/screens/meditation/components/controls/components/button/button/button.tsx]: Can we reuse exsiting button without creating new one?

**Review by v0ldemar01 — CHANGES_REQUESTED** [review]

File: mobile/index.js
Comment (liza-veis) [inline mobile/index.js]: Won't it work without this handler?

or what if we try to pass empty function here 
```suggestion
TrackPlayer.registerPlaybackService(() => async () => {});
```

File: mobile/src/libs/components/app/app.tsx
Comment (liza-veis) [inline mobile/src/libs/components/app/app.tsx]: let's follow the flow as with notifications - create a separate package, then create a meditation dir in slices, add initPlayer action here and the dispatch it in meditation screen

File: mobile/src/libs/components/header/header.tsx
Comment (liza-veis) [inline mobile/src/libs/components/header/header.tsx]:
```suggestion
```

File: mobile/src/libs/components/header/header.tsx
Comment (liza-veis) [inline mobile/src/libs/components/header/header.tsx]:
```suggestion
```

File: mobile/src/libs/hooks/use-track-player-controls/use-track-player-controls.hook.ts
Comment (liza-veis) [inline mobile/src/libs/hooks/use-track-player-controls/use-track-player-controls.hook.ts]:
```suggestion
    handlePlayPause,
```

File: mobile/src/libs/hooks/use-track-player-progress/use-track-player-progress.hook.ts
Comment (liza-veis) [inline mobile/src/libs/hooks/use-track-player-progress/use-track-player-progress.hook.ts]: we don't use function notation

File: mobile/src/screens/meditation-list/components/meditation-item/meditation-item.tsx
Comment (liza-veis) [inline mobile/src/screens/meditation-list/components/meditation-item/meditation-item.tsx:35]: we shouldn't pass playlists, we will have them in slice

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: mobile/src/libs/components/header/header.tsx
Comment (danilittaa) [reply mobile/src/libs/components/header/header.tsx]: can't do it 
image

File: mobile/src/libs/hooks/use-track-player-controls/use-track-player-controls.hook.ts
Comment (danilittaa) [reply mobile/src/libs/hooks/use-track-player-controls/use-track-player-controls.hook.ts]: If I do it like this, I get an error:  error  Promise-returning function provided to property where a void return was 
expected  @typescript-eslint/no-misused-promises

File: mobile/src/libs/components/header/header.tsx
Comment (liza-veis) [reply mobile/src/libs/components/header/header.tsx]: don't you return one element? if so this should work
```
if (isVisible) {
  return <View>{children}</View>
}

return <View>{children}</View>
```

File: mobile/src/libs/components/header/header.tsx
Comment (liza-veis) [inline mobile/src/libs/components/header/header.tsx]: what is the difference between isVisible and isArrowVisible? do we need all of these? why it should be with position absolute?

File: mobile/src/screens/meditation/meditation.tsx
Comment (liza-veis) [inline mobile/src/screens/meditation/meditation.tsx]: we should have a player component and all of this logic including controls should be inside it

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: mobile/src/libs/helpers/helpers.ts
Comment (dmytro-kucherenko) [inline mobile/src/libs/helpers/helpers.ts:5]: `.js` is redundant on mobile

File: mobile/src/libs/components/header/header.tsx
Comment (danilittaa) [reply mobile/src/libs/components/header/header.tsx]: isVisible is used to make the header completely transparent (as it should be for the player page), while isArrowVisible is specifically responsible for the back arrow. 

isArrowVisible should have 'position: 'absolute'' because on some pages, the text in the header should be centered. Using 'justifyContent: 'center',' would center both the text and the arrow, but the arrow should always be on the left. 

isVisible has 'position: absolute' because the page uses a gradient, and without 'position: 'absolute'', the header would have its own color, which goes against the design in Figma

Comment (liza-veis) [reply mobile/src/libs/components/header/header.tsx]: maybe something like this will work https://medium.com/@athletecoder/react-navigation-gradient-background-fc8d929386df. It's better not to make it position absolute, but to try to set gradient background on navigation level

File: mobile/src/libs/types/meditation/meditation-track.type.ts
Comment (liza-veis) [inline mobile/src/libs/types/meditation/meditation-track.type.ts]: we should not use library types in our app, we will only use type for  MeditationEntryItem that we will get from backend and then we will map to properties we need only in place where we use this library, so you can leave comment like 'TODO: delete this type when backend part is ready'

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: mobile/src/libs/components/button/button.tsx
Comment (fshabanov) [inline mobile/src/libs/components/button/button.tsx]: Let's separate 'rounded' from others. Rounded is more about the shape of the button, but others are more about the color

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: mobile/src/libs/components/player/player.tsx
Comment (v0ldemar01) [inline mobile/src/libs/components/player/player.tsx]: Can we simplify it?

File: mobile/src/screens/meditation/meditation.tsx
Comment (v0ldemar01) [inline mobile/src/screens/meditation/meditation.tsx]: Let's move to constant in component

**Review by v0ldemar01 — CHANGES_REQUESTED** [review]

File: mobile/src/screens/meditation/meditation.tsx
Comment (OlesiaTr) [inline mobile/src/screens/meditation/meditation.tsx]: Is it supposed to be PLayer or just typo?

Comment (danilittaa) [reply mobile/src/screens/meditation/meditation.tsx]: it's typo, thank you)

**Review by OlesiaTr — APPROVED** [review]

**Review by v0ldemar01 — APPROVED** [review]

**Review by liza-veis — APPROVED** [review]

**Pull request merged**
